docs/upgrade.html

```diff
@@ -91,6 +91,7 @@
 
 <ul>
     <li> Starting from Kafka 0.10.0.0, a new client library named <b>Kafka Streams</b> is available for stream processing on data stored in Kafka topics. This new client library only works with 0.10.x and upward versioned brokers due to message format changes mentioned above. For more information please read <a href="#streams_overview">this section</a>.</li>
+    <li> The new producer will use compression scheme's default block size per-partition to replace the universal hard-coded block size of 1KB in order to achieve better compression ratio; for Snappy specifically this means the block size will increase to 32KB, hence increasing JVM heap usage.</li>
```

Comment:
Maybe:

> If compression with snappy or gzip is enabled, the new producer will use the compression scheme's default buffer size (this is already the case for LZ4) instead of 1 KB in order to improve the compression ratio. Note that the default buffer sizes for snappy, gzip and LZ4 are 0.5 KB, 32 KB and 64KB respectively. For the snappy case, a producer with 5000 partitions will require an additional 155 MB of JVM heap.